### PR TITLE
test(healthcare): include full operation in exception message

### LINKED ISSUE #7433: healthcare.api-client.v1beta1.fhir.fhir_stores_test: test_create_delete_fhir_store failed
Note: #6707 was also for this test, but it was closed more than 10 days ago. So, I didn't mark it flaky.

----

commit: 8d48cd7a2b6c15a84de1cd7ef2fb72b4550fa799
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/01ebeef5-9c39-4018-a953-400ea44e5402), [Sponge](http://sponge2/01ebeef5-9c39-4018-a953-400ea44e5402)
status: failed
<details><summary>Test output</summary><br><pre>Traceback (most recent call last):
  File "/workspace/healthcare/api-client/v1beta1/fhir/fhir_stores_test.py", line 72, in test_create_delete_fhir_store
    create()
  File "/workspace/healthcare/api-client/v1beta1/fhir/.nox/py-3-10/lib/python3.10/site-packages/backoff/_sync.py", line 94, in retry
    ret = target(*args, **kwargs)
  File "/workspace/healthcare/api-client/v1beta1/fhir/fhir_stores_test.py", line 69, in create
    fhir_stores.create_fhir_store(
  File "/workspace/healthcare/api-client/v1beta1/fhir/fhir_stores.py", line 106, in create_fhir_store
    response = request.execute()
  File "/workspace/healthcare/api-client/v1beta1/fhir/.nox/py-3-10/lib/python3.10/site-packages/googleapiclient/_helpers.py", line 131, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/workspace/healthcare/api-client/v1beta1/fhir/.nox/py-3-10/lib/python3.10/site-packages/googleapiclient/http.py", line 937, in execute
    raise HttpError(resp, content, uri=self.uri)
googleapiclient.errors.HttpError: <HttpError 404 when requesting https://healthcare.googleapis.com/v1beta1/projects/python-docs-samples-tests/locations/us-central1/datasets/test_dataset_237e3132-41d8-472a-8260-8e85a408d49f/fhirStores?fhirStoreId=test_fhir_store-89e80b74-ed82-4a51-90cf-04e837447b21&alt=json returned "dataset not initialized". Details: "dataset not initialized"></pre></details>

### PR DESCRIPTION
**Pull request by busunkim96**
## Description

For #7433.

https://github.com/GoogleCloudPlatform/python-docs-samples/issues/7433#issuecomment-1053561512 These tests failed with a `KeyError` yesterday in the 3.7 periodic build last night:

```
Traceback (most recent call last):
  File "/workspace/healthcare/api-client/v1beta1/fhir/fhir_stores_test.py", line 62, in test_dataset
    wait_for_operation(operation["name"])
  File "/workspace/healthcare/api-client/v1beta1/fhir/.nox/py-3-10/lib/python3.10/site-packages/google/api_core/retry.py", line 283, in retry_wrapped_func
    return retry_target(
  File "/workspace/healthcare/api-client/v1beta1/fhir/.nox/py-3-10/lib/python3.10/site-packages/google/api_core/retry.py", line 190, in retry_target
    return target()
  File "/workspace/healthcare/api-client/v1beta1/fhir/fhir_stores_test.py", line 51, in wait_for_operation
    if not operation["done"]:
KeyError: 'done'
```

I was unable to reproduce the error locally using the CI project, and I'm unsure what is causing the frequent failures on the periodic build. 

This PR prints out the full `operation` when the `done` key is missing. It will not resolve #7433, but should be helpful for determining the root case.

### CODE REVIEW

File: healthcare/api-client/v1beta1/fhir/fhir_resources_test.py [hunk 1]
```diff
@@ -58,9 +58,19 @@ def wait_for_operation(operation_name: str):
...
+            raise OperationNotComplete()
+    except KeyError:
+        # NOTE(busunkim): An operation should always have the 'done' field,
+        # but periodic logs indicate it is sometimes missing.
+        # This code prints out the full operation response for debugging purposes.
+        # https://github.com/GoogleCloudPlatform/python-docs-samples/issues/7433#issuecomment-1053561512
+        raise Exception(
+            "Operation missing key 'done':",
+            operation
+        )
```
Comment (parthea) [inline healthcare/api-client/v1beta1/fhir/fhir_resources_test.py:74]:
```suggestion


```

File: healthcare/api-client/v1beta1/fhir/fhir_stores_test.py [hunk 2]
```diff
@@ -48,9 +48,19 @@ def wait_for_operation(operation_name: str):
...
+            raise OperationNotComplete()
+    except KeyError:
+        # NOTE(busunkim): An operation should always have the 'done' field,
+        # but periodic logs indicate it is sometimes missing.
+        # This code prints out the full operation response for debugging purposes.
+        # https://github.com/GoogleCloudPlatform/python-docs-samples/issues/7433#issuecomment-1053561512
+        raise Exception(
+            "Operation missing key 'done':",
+            operation
+        )
```
Comment (parthea) [inline healthcare/api-client/v1beta1/fhir/fhir_stores_test.py:64]:
```suggestion


```

**Review by parthea — APPROVED** [review]
LGTM

**Review by noerog — APPROVED** [review]

**Comment by busunkim96** [comment]
Thanks for fixing the lint issues @parthea!

By a stroke of luck, the build failed on the most 3.7 presubmit run: I'll share this with the Healthcare API folks.

```py
Traceback (most recent call last):
  File "/workspace/healthcare/api-client/v1/fhir/fhir_stores_test.py", line 69, in wait_for_operation
    if not operation["done"]:
KeyError: 'done'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/workspace/healthcare/api-client/v1/fhir/fhir_stores_test.py", line 87, in test_dataset
    wait_for_operation(operation["name"])
  File "/workspace/healthcare/api-client/v1/fhir/.nox/py-3-7/lib/python3.7/site-packages/google/api_core/retry.py", line 288, in retry_wrapped_func
    on_error=on_error,
  File "/workspace/healthcare/api-client/v1/fhir/.nox/py-3-7/lib/python3.7/site-packages/google/api_core/retry.py", line 190, in retry_target
    return target()
  File "/workspace/healthcare/api-client/v1/fhir/fhir_stores_test.py", line 78, in wait_for_operation
    operation
Exception: ("Operation missing key 'done':", {'name': 'projects/python-docs-samples-tests-py37/locations/us-central1/datasets/test_dataset_ec158b3a-7237-449f-9d82-300ffffcf93e/operations/8434283371143626753', 'metadata': {'@type': 'type.googleapis.com/google.cloud.healthcare.v1.OperationMetadata', 'apiMethodName': 'google.cloud.healthcare.v1.dataset.DatasetService.CreateDataset', 'createTime': '2022-02-28T23:27:17.422883Z', 'logsUrl': 'https://console.cloud.google.com/logs/viewer?advancedFilter=operation.id%3Dprojects%2Fpython-docs-samples-tests-py37%2Flocations%2Fus-central1%2Fdatasets%2Ftest_dataset_ec158b3a-7237-449f-9d82-300ffffcf93e%2Foperations%2F8434283371143626753%0Atimestamp+%3E%3D+%222022-02-28T23%3A27%3A17Z%22&project=python-docs-samples-tests-py37&interval=NO_LIMIT'}})
```

File: healthcare/api-client/v1/fhir/fhir_resources_test.py [hunk 3]
```diff
@@ -61,8 +61,9 @@ def wait_for_operation(operation_name: str):
         .get(name=operation_name)
         .execute()
     )
-    if not operation["done"]:
-        raise OperationNotComplete()
+
+    if not operation.get("done", False):
+        raise OperationNotComplete(operation)
```
Comment (busunkim96) [inline healthcare/api-client/v1/fhir/fhir_resources_test.py:67]: I've tweaked this to assume if "done" is not present, `done = False`. The exception message will contain the operation.

**Comment by busunkim96** [comment]
I've made some changes since your review, @parthea @noerog could you take another look?

**Review by parthea — APPROVED** [review]

**Pull request merged**